cirq-pasqal/cirq_pasqal/pasqal_device.py

```diff
@@ -97,36 +97,39 @@ def is_pasqal_device_op(self, op: cirq.ops.Operation) -> bool:
         if not isinstance(op, cirq.ops.Operation):
             raise ValueError('Got unknown operation:', op)
 
-        valid_op = isinstance(
-            op.gate,
+        if isinstance(op.gate, cirq.ops.MeasurementGate):
+            return True
+
+        op_gate = op.gate.sub_gate if isinstance(op.gate, cirq.ops.ParallelGate) else op.gate
+
+        if isinstance(
+            op_gate,
             (
                 cirq.ops.IdentityGate,
-                cirq.ops.MeasurementGate,
```

Comment:
Here, too, not sure I understand why there's a special case for `MeasurementGate`. I get that it's somewhat tricky to define a parallel measurement gate in general because it could mean that the key is repeated, but what if the `ParallelGate` contained exactly one `MeasurementGate`? Not clear to me that we should treat this differently here.

Reply:
Because of the trickiness in defining the a parallel measurement gate, I err'ed on the cautious side of treating measurement gates separately. 

If a `ParallelGate` contains exactly one `MeasurementGate`, we won't recognize it as a known gate here and it'll get decomposed into a single `MeasurementGate` and hence work as expected. But if we do treat the parallel measurement gate's in an identical fashion, we could run into other issues like repeated keys, confusing semantics because `MeasurementGate` itself accepts a `num_qubits` argument which specifies the number of qubits to act on, and I'm also not sure whether the hardware actually supports a single operation for doing parallel measurements or not, similar to parallel single qubit rotation gates. 

In the end, this decision should be made by the device authors. I'll be happy to change if there's consensus that it should be changed.

Reply:
Sounds good. We might consider a refactor to make it explicitly clear that ParallelGate with MeasurementGate inside is disallowed, e.g. something like:
```python
if isinstance(op.gate, cirq.MeasurementGate):
    return True
if isinstance(op.gate, cirq.ParallelGate) and isinstance(op.gate.subgate, cirq.MeasurementGate):
    return False
```

Reply:
done